src/timer.c

```diff
@@ -36,5 +34,11 @@ void mg_timer_poll(struct mg_timer **head, uint64_t now_ms) {
       t->fn(t->arg);
     }
     t->flags |= MG_TIMER_CALLED;
+
+    // If this timer is not repeating and marked AUTODELETE, remove it
+    if (!(t->flags & MG_TIMER_REPEAT) && (t->flags & MG_TIMER_AUTODELETE)) {
+      mg_timer_free(head, t);
+      free(t);
+    }
   }
```

Comment:
One day we should resume our brainstorm... this Timer API has no way to re-use a single-use timer. That could avoid unnecessary memory fragmentation.
OTOH, it has served well for a long time and there are very few, if any, uses of run-once timers in Mongoose code.